client/jetpack-connect/happychat-button.jsx

```diff
@@ -13,20 +13,30 @@ import { localize } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
+
+import analytics from 'lib/analytics';
 import HappychatButton from 'components/happychat/button';
 import HappychatConnection from 'components/happychat/connection';
 import { isEnabled } from 'config';
 import { getCurrentUserId } from 'state/current-user/selectors';
 import hasActiveHappychatSession from 'state/happychat/selectors/has-active-happychat-session';
 import isHappychatAvailable from 'state/happychat/selectors/is-happychat-available';
 
+const getHappyChatButtonClickHandler = eventName => {
+	eventName = eventName || 'calypso_jpc_chat_initiated';
+	return () => {
+		analytics.tracks.recordEvent( eventName );
```

Comment:
I'd recommend using the Redux `recordTracksEvent` instead of `analytics.tracks.recordEvent`.

Reply:
well, this is a method that is not connected ... it could be, but it would complicate the code a little bit ... I think I would rather prefer to leave it as is, if it doesn't have any problem I'm not seeing

Reply:
Actually, it's more about code consistency - AFAIK we try to use the Redux analytics actions rather than the lib directly for any new code. 

Personally I'd prefer refactoring `JetpackConnectHappychatButton` to a class and making `getHappyChatButtonClickHandler` a method of that class, so we can use the `recordTracksEvent` action. But I'll leave this to your judgement.